### PR TITLE
fix(issue-platform): Accept new `ProfileFileIOGroupType` issue type

### PR DESCRIPTION
**Pull request by wedamija**
In https://github.com/getsentry/sentry/pull/44913 we added new issue types, but also the profile team stopped sending type 2000, so we stopped ingesting all profile issues. Fixed that here.

### CODE REVIEW

File: src/sentry/issues/occurrence_consumer.py [hunk 1]
```diff
@@ -264,8 +265,14 @@ def _process_message(
             txn.set_tag("project_id", project.id)
             txn.set_tag("project_slug", project.slug)
 
-            if not features.has("organizations:profile-blocked-main-thread-ingest", organization):
-                metrics.incr("occurrence_ingest.dropped_feature_disabled", sample_rate=1.0)
+            if occurrence_data["type"] not in PROFILE_FILE_IO_ISSUE_TYPES or not features.has(
+                "organizations:profile-blocked-main-thread-ingest", organization
+            ):
```
Comment (rjo100) [inline src/sentry/issues/occurrence_consumer.py:270]: are we using `"organizations:profile-blocked-main-thread-ingest"` as a multifunctional feature flag here? does `ProfileFileIOGroupType` have its own flag?

**Review by rjo100 — APPROVED** [review]
looks good just one Q

File: src/sentry/issues/occurrence_consumer.py [hunk 2]
```diff
@@ -264,8 +265,14 @@ def _process_message(
             txn.set_tag("project_id", project.id)
             txn.set_tag("project_slug", project.slug)
 
-            if not features.has("organizations:profile-blocked-main-thread-ingest", organization):
-                metrics.incr("occurrence_ingest.dropped_feature_disabled", sample_rate=1.0)
+            if occurrence_data["type"] not in PROFILE_FILE_IO_ISSUE_TYPES or not features.has(
+                "organizations:profile-blocked-main-thread-ingest", organization
```
Comment (barkbarkimashark) [inline src/sentry/issues/occurrence_consumer.py:269]: Is `profile-blocked-main-thread-ingest` flag being used for `ProfileFileIOGroupType` as well?

File: src/sentry/issues/occurrence_consumer.py (hunk 1 above)
Comment (wedamija) [reply src/sentry/issues/occurrence_consumer.py:270]: We're using the same feature flag here. It'd be a lot of work at the moment to go through and manually add a new flag for each type. `ProfileFileIOGroupType` is actually the same data from `ProfileBlockedThreadGroupType`, profiling just made it more specific. Not sure if we're even going to get type 2000 anymore.

Comment (rjo100) [reply src/sentry/issues/occurrence_consumer.py:262]: right. got it. just asking for my own understanding mostly

**Pull request merged**